src/extension/rn-extension.ts

```diff
@@ -83,13 +83,18 @@ export function activate(context: vscode.ExtensionContext): void {
     });
 }
 
-export function deactivate(): void {
-    // Kill any packager processes that we spawned
-    entryPointHandler.runFunction("extension.deactivate",
-        ErrorHelper.getInternalError(InternalErrorCode.FailedToStopPackagerOnExit),
-        () => {
-            commandPaletteHandler.stopPackager();
-        }, /*errorsAreFatal*/ true);
+export function deactivate(): Q.Promise<void> {
+    return Q.Promise<void>(function (resolve) {
+        // Kill any packager processes that we spawned
+        entryPointHandler.runFunction("extension.deactivate",
+            ErrorHelper.getInternalError(InternalErrorCode.FailedToStopPackagerOnExit),
+            () => {
+                commandPaletteHandler.stopPackager().done(() => {
+                    // Tell vscode that we are done with deactivation
+                    resolve(undefined);
```

Comment:
Looks like [void 0](https://github.com/Microsoft/vscode-react-native/search?utf8=%E2%9C%93&q=resolve%28void+0%29&type=Code) is used for this case - what do you think?